frontend/packages/console-app/src/components/nodes/NodeTerminal.tsx

```diff
@@ -0,0 +1,169 @@
+import { Alert } from '@patternfly/react-core';
+import {
+  Firehose,
+  FirehoseResource,
+  FirehoseResult,
+  LoadingBox,
+} from '@console/internal/components/utils';
+import { NodeKind, PodKind } from '@console/internal/module/k8s';
+import { getRandomChars } from '@console/shared/src/utils';
+import { PodExecLoader } from '../../../../../public/components/pod';
+import { ImageStreamTagModel, PodModel } from '../../../../../public/models';
+import { k8sCreate, k8sGet, k8sKillByName } from '../../../../../public/module/k8s';
+
+type NodeTerminalErrorProps = {
+  error: React.ReactNode;
+};
+
+type NodeTerminalInnerProps = {
+  obj?: FirehoseResult<PodKind>;
+};
+
+type NodeTerminalProps = {
+  obj: NodeKind;
+};
+
+const getDebugImage = async (): Promise<string> => {
+  try {
+    const istag = await k8sGet(ImageStreamTagModel, 'tools:latest', 'openshift');
+    return istag.image.dockerImageReference;
+  } catch (e) {
+    return 'registry.redhat.io/rhel7/support-tools';
+  }
+};
+
+const getDebugPod = async (name: string, namespace: string, nodeName: string): Promise<PodKind> => {
+  const image = await getDebugImage();
+  return {
+    kind: 'Pod',
+    apiVersion: 'v1',
+    metadata: {
+      name,
+      namespace,
+      annotations: {
+        'debug.openshift.io/source-container': 'container-00',
+        'debug.openshift.io/source-resource': `/v1, Resource=nodes/${nodeName}`,
+      },
+    },
+    spec: {
+      activeDeadlineSeconds: 21600,
+      volumes: [
+        {
+          name: 'host',
+          hostPath: {
+            path: '/',
+            type: 'Directory',
+          },
+        },
+      ],
+      containers: [
+        {
+          name: 'container-00',
+          image,
+          command: ['/bin/sh'],
+          resources: {},
+          volumeMounts: [
+            {
+              name: 'host',
+              mountPath: '/host',
+            },
+          ],
+          securityContext: {
+            privileged: true,
+            runAsUser: 0,
+          },
+          stdin: true,
+          stdinOnce: true,
+          tty: true,
+        },
+      ],
+      restartPolicy: 'Never',
+      nodeName,
+      hostNetwork: true,
+      hostPID: true,
+    },
+  };
+};
+
+const NodeTerminalError: React.FC<NodeTerminalErrorProps> = ({ error }) => {
+  return (
+    <div className="co-m-pane__body">
+      <Alert variant="danger" isInline title={error} />
+    </div>
+  );
+};
+
+const NodeTerminalInner: React.FC<NodeTerminalInnerProps> = ({ obj }) => {
+  const message = (
+    <p>
+      To use host binaries, run <code>chroot /host</code>
+    </p>
+  );
+  switch (obj?.data?.status?.phase) {
+    case 'Failed':
+      return (
+        <NodeTerminalError
+          error={
+            <>
+              The debug pod failed.{' '}
+              {obj?.data?.status?.containerStatuses[0]?.state?.terminated &&
+                obj?.data?.status?.containerStatuses[0]?.state?.terminated?.message}
```

Comment:
The `?` in the second statement aren't needed since you tested for `terminated` above. But that check shouldn't be needed.

```suggestion
              {obj?.data?.status?.containerStatuse?.[0]?.state?.terminated?.message}
```

Reply:
Thanks, @spadgett.  Updated.